### PR TITLE
The azimuthally-averaged and spatially-resolved Kennicutt-Schmidt relations

### PR DESCRIPTION
**Pull request by Fonotec**
This is all the data that we need in order to compare with both the neutral (molecular + atomic) and molecular KS relations, both for the azimuthally-averaged and spatially-resolved KS relations. 

Currently, I made my first convert script for the Schruba 2011 dataset. @bwvdnbro let me know any current suggestions to improve the scripts.

After this script I will make the following scripts:
- The azimuthally-averaged neutral KS relation
- The spatially-resolved neutral KS relation
- The spatially-resolved molecular KS relation

### CODE REVIEW

**Review by bwvdnbro — CHANGES_REQUESTED** [review]
I have left an extensive list of comments in one of your scripts. The same comments apply to the other scripts.

Generally:
 - use `scipy.stats.binned_statistic`, it is much faster than any binning algorithm you can come up with yourself
 - try to explain where non trivial conversion factors (e.g. the `1.36` factor) come from
 - use the bin size as scatter for the x variable

**Comment by bwvdnbro** [comment]
Another thing: could you run the formatting script? As you can see, the CI fails.

File: data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+processed = ObservationalData()
+
+comment = "Based on the IRAM HERCULES survey."
+citation = "Schruba et al. (2011)"
+bibcode = "2011AJ....142...37S"
+name = "azimuthally-averaged H2 Gas Surface Density vs Star Formation Rate Surface Density"
+plot_as = "points"
+
+sigma_SFR, sigma_SFR_err, sigma_HI, sigma_HI_err, sigma_H2, sigma_H2_err, quality = np.genfromtxt(input_filename, unpack=True, usecols=(3,4,5,6,7,8,10))
```
[the remark this answers is not in the document]
Comment (Fonotec) [reply data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py]: Okay, I fixed this in all the files.

File: data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+array_of_interest = np.arange(-1,3,0.25)
+minimum_surface_density = 0.0
+
+Obs_HI = (sigma_HI)/1.36
+Obs_H2 = (sigma_H2)/1.36
```
[the remark this answers is not in the document]
Comment (Fonotec) [reply data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py]: I added a comment in each file regarding this, similarly, for the optional factor for Sigma_SFR I also added a comment.

File: data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+from velociraptor.observations.objects import ObservationalData
+
+import unyt
+import numpy as np
+import os
+import sys
+import matplotlib.pyplot as plt
+
+def bin_data_general(array_x, array_y, array_x_bin, x_limit, print_stuff=False):
```
[the remark this answers is not in the document]
Comment (Fonotec) [reply data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py]: I added this improvement to the code

File: data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            print(array_y[mask])
+            print(stats.describe(array_y[mask]))
+        try:
+            y_array_bin_std_up[i], y_array_bin_std_down[i] = np.transpose(np.percentile(array_y[mask], [16,84]))
+        except:
+            y_array_bin_std_up[i], y_array_bin_std_down[i] = [0., 0.]
+
+    array_x_bin =  (array_x_bin[1:] + array_x_bin[:-1])/2.
+    y_array_bin_std_up = np.abs(y_array_bin_std_up - y_array_bin)
+    y_array_bin_std_down = np.abs(y_array_bin_std_down - y_array_bin)
```
[the remark this answers is not in the document]
Comment (Fonotec) [reply data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py]: That is true, I added this to the code

File: data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+binned_data = bin_data_general(np.log10(Obs_H2), np.log10(Obs_SFR), array_of_interest, minimum_surface_density)
+
+SigmaH2 = unyt.unyt_array(10**binned_data[0], units="Msun/pc**2")
+
+SigmaSFR = unyt.unyt_array(10**binned_data[1], units="Msun/yr/kpc**2")
+
+SigmaSFR_err = unyt.unyt_array([10**(binned_data[1]) - 10**(binned_data[1]-binned_data[3]), 10**(binned_data[1]+binned_data[2]) -10**(binned_data[1])], units="Msun/yr/kpc**2")
+
+processed.associate_x(SigmaH2, scatter=None, comoving=False, description="H2 Surface density")
```
[the remark this answers is not in the document]
Comment (Fonotec) [reply data/AzimuthallyAveragedMolecularKSRelation/conversion/convertSchruba2011.py]: True, I added this change to the code

**Review by bwvdnbro — APPROVED** [review]

**Comment by bwvdnbro** [comment]
Looks good. I noticed that the CI was not working properly, but I couldn't push to your fork. So I have fixed this and merged this PR manually.

**Pull request merged**